transport/ProbeServer.cpp

```diff
@@ -0,0 +1,224 @@
+#include "transport/ProbeServer.h"
+#include "logger/Logger.h"
+#include "transport/ice/Stun.h"
+#include "utils/Time.h"
+#include <unistd.h>
+
+namespace transport
+{
+ProbeServer::ProbeServer(ice::IceConfig& iceConfig) : _iceConfig(iceConfig)
+{
+    // TODO copied from IceSession.cpp; refactor
+    char ufrag[14 + 1];
+    char pwd[24 + 1]; // length selected to make attribute *4 length
+
+    generateCredentialString(ufrag, sizeof(ufrag) - 1);
+    generateCredentialString(pwd, sizeof(pwd) - 1);
+
+    _probeCredentials = std::make_pair<std::string, std::string>(ufrag, pwd);
+}
+
+// Endpoint::IEvents
+void ProbeServer::onRtpReceived(Endpoint& endpoint,
+    const SocketAddress& source,
+    const SocketAddress& target,
+    memory::UniquePacket packet)
+{
+}
+
+void ProbeServer::onDtlsReceived(Endpoint& endpoint,
+    const SocketAddress& source,
+    const SocketAddress& target,
+    memory::UniquePacket packet){};
+
+void ProbeServer::onRtcpReceived(Endpoint& endpoint,
+    const SocketAddress& source,
+    const SocketAddress& target,
+    memory::UniquePacket packet)
+{
+}
+
+void ProbeServer::onIceReceived(Endpoint& endpoint,
+    const SocketAddress& source,
+    const SocketAddress& target,
+    memory::UniquePacket packet)
+{
+    replyStunOk(endpoint, source, std::move(packet));
+}
+
+void ProbeServer::onRegistered(Endpoint& endpoint)
+{
+    const std::lock_guard<std::mutex> locker(_endpointsLock);
+
+    if (endpoint.getTransportType() != ice::TransportType::UDP)
+    {
+        return;
+    }
+
+    auto port = endpoint.getLocalPort().getPort();
```

Comment:
This will not work. SMB can open the same port on multiple interfaces.

Reply:
I removed lists of UDP/TCP ports and instead keep single ICE candidates list (vector).